server/src/internalClusterTest/java/org/opensearch/ratelimitting/admissioncontrol/AdmissionForClusterManagerIT.java

```diff
@@ -0,0 +1,182 @@
+    private String datanode;
+    private ResourceUsageCollectorService cMResourceCollector;
+
+    private static final Settings DISABLE_ADMISSION_CONTROL = Settings.builder()
+        .put(ADMISSION_CONTROL_TRANSPORT_LAYER_MODE.getKey(), AdmissionControlMode.DISABLED.getMode())
+        .build();
+
+    private static final Settings ENFORCE_ADMISSION_CONTROL = Settings.builder()
+        .put(ResourceTrackerSettings.GLOBAL_CPU_USAGE_AC_WINDOW_DURATION_SETTING.getKey(), TimeValue.timeValueMillis(500))
+        .put(ADMISSION_CONTROL_TRANSPORT_LAYER_MODE.getKey(), AdmissionControlMode.ENFORCED)
+        .put(CLUSTER_ADMIN_CPU_USAGE_LIMIT.getKey(), 50)
+        .build();
+
+    @Before
+    public void init() {
+        String clusterManagerNode = internalCluster().startClusterManagerOnlyNode(
+            Settings.builder().put(DISABLE_ADMISSION_CONTROL).build()
+        );
+        datanode = internalCluster().startDataOnlyNode(Settings.builder().put(DISABLE_ADMISSION_CONTROL).build());
+
+        ensureClusterSizeConsistency();
+        ensureGreen();
+
+        // Disable the automatic resource collection
+        clusterManagerNodeId = internalCluster().clusterService(clusterManagerNode).localNode().getId();
+        cMResourceCollector = internalCluster().getClusterManagerNodeInstance(ResourceUsageCollectorService.class);
+        cMResourceCollector.stop();
+
+        // Enable admission control
+        client().admin().cluster().prepareUpdateSettings().setTransientSettings(ENFORCE_ADMISSION_CONTROL).execute().actionGet();
+    }
+
+    public void testAdmissionControlEnforced() throws Exception {
+        cMResourceCollector.collectNodeResourceUsageStats(clusterManagerNodeId, System.currentTimeMillis(), 97, 99, new IoUsageStats(98));
+
+        // Write API on ClusterManager
+        assertAcked(prepareCreate("test").setMapping("field", "type=text").setAliases("{\"alias1\" : {}}"));
+
+        // Read API on ClusterManager
+        GetAliasesRequest aliasesRequest = new GetAliasesRequest();
+        aliasesRequest.aliases("alias1");
+        try {
+            dataNodeClient().admin().indices().getAliases(aliasesRequest).actionGet();
+            fail("expected failure");
+        } catch (Exception e) {
+            assertTrue(e instanceof OpenSearchRejectedExecutionException);
+        }
+
+        client().admin().cluster().prepareUpdateSettings().setTransientSettings(DISABLE_ADMISSION_CONTROL).execute().actionGet();
+        GetAliasesResponse getAliasesResponse = dataNodeClient().admin().indices().getAliases(aliasesRequest).actionGet();
+        assertThat(getAliasesResponse.getAliases().get("test").size(), equalTo(1));
+
+        AdmissionControlService admissionControlServicePrimary = internalCluster().getClusterManagerNodeInstance(
+            AdmissionControlService.class
+        );
+        AdmissionControllerStats admissionStats = admissionControlServicePrimary.stats().getAdmissionControllerStatsList().get(0);
```

Comment:
I am assuming these stats are reset with every test as in init it is starting new nodes.

Reply:
Yes, ResourceCollector is intialized on start;